ws-server/src/tests.rs

```diff
@@ -58,6 +58,19 @@ async fn single_method_call_with_params_works() {
 	assert_eq!(response, ok_response(JsonValue::Number(3.into()), Id::Num(1)));
 }
 
+#[tokio::test]
+async fn single_method_call_with_faulty_params_returns_err() {
+	let _ = env_logger::try_init();
+	let (server_started_tx, server_started_rx) = oneshot::channel::<SocketAddr>();
+	tokio::spawn(server(server_started_tx));
+	let server_addr = server_started_rx.await.unwrap();
+	let mut client = WebSocketTestClient::new(server_addr).await.unwrap();
+
+	let req = r#"{"jsonrpc":"2.0","method":"add", "params":["Invalid"],"id":1}"#;
```

Comment:
Would be good to cover a few more odd cases, e.g. `{"jsonrpc":"2.0","method":"add", "params":["Invalid"],"id":"hey ho"}`, `{"jsonrpc":"2.0","method":"add", "params": "Invalid", "id":1}`, `{"id":1}` etc

Reply:
those would simplify be invalid `JSON-RPC requests`, seems unrelated we already got tests for this, except params  is not array or object.

Reply:
I get where the motivation for those tests but more related to #276 than this PR, will not address that here.